server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -680,6 +680,13 @@ public Discoverer getMatchingDiscover(final Hypervisor.HypervisorType hypervisor
             }
         }
 
+        if ((hypervisorType.equalsIgnoreCase(HypervisorType.BareMetal.toString()))) {
+            if (hostTags == null || hostTags.size() == 0) {
```

Comment:
Hey @harikrishna-patnala just a little tip on line 684. You could use Apache Commons' CollectionUtils to make such checks, it does exactly the same thing but is a little more clean. But that's just a little adjustment. http://commons.apache.org/proper/commons-collections/javadocs/api-release/org/apache/commons/collections4/CollectionUtils.html#isEmpty(java.util.Collection)